posthog/session_recordings/queries/session_replay_events.py

```diff
@@ -189,72 +189,49 @@ def get_events_for_session(self, session_id: str, team: Team, limit: int = 100)
-
-        # Build event pattern string for comparison
-        target_pattern = ",".join([e["event"] for e in target_events])
-
         query = """
+        WITH target_urls AS (
+            SELECT groupArrayArray(all_urls) as url_sequence
+            FROM session_replay_events
+            WHERE team_id = %(team_id)s
+            AND session_id = %(session_id)s
+        )
         SELECT
             sre.session_id,
             sre.distinct_id,
-            min_first_timestamp,
-            max_last_timestamp,
+            min_first_timestamp as start_time,
+            max_last_timestamp as end_time,
             click_count,
             keypress_count,
             mouse_activity_count,
             active_milliseconds,
-            arrayStringConcat(groupArray(e.event), ',') as event_pattern
+            all_urls as urls,
+            target_urls.url_sequence
         FROM session_replay_events sre
-        JOIN events e ON e.team_id = sre.team_id AND e.$session_id = sre.session_id
+        CROSS JOIN target_urls
         WHERE sre.team_id = %(team_id)s
         AND sre.session_id != %(session_id)s
         GROUP BY
             sre.session_id,
             sre.distinct_id,
-            sre.min_first_timestamp,
-            sre.max_last_timestamp,
-            sre.click_count,
-            sre.keypress_count,
-            sre.mouse_activity_count,
-            sre.active_milliseconds
+            min_first_timestamp,
+            max_last_timestamp,
+            click_count,
+            keypress_count,
+            mouse_activity_count,
+            active_milliseconds,
+            all_urls,
+            target_urls.url_sequence
         HAVING
-            -- Similar activity levels (within similarity_range)
-            abs(click_count - %(target_clicks)s) <= greatest(%(target_clicks)s * %(similarity_range)s, 5) AND
-            abs(keypress_count - %(target_keypresses)s) <= greatest(%(target_keypresses)s * %(similarity_range)s, 5) AND
-            abs(mouse_activity_count - %(target_mouse)s) <= greatest(%(target_mouse)s * %(similarity_range)s, 5) AND
-            abs(active_milliseconds - %(target_active_ms)s) <= greatest(%(target_active_ms)s * %(similarity_range)s, 5000) AND
-            -- Similar event pattern using Levenshtein distance
-            length(event_pattern) > 0 AND
-            event_pattern != %(target_pattern)s AND
-            length(event_pattern) >= greatest(length(%(target_pattern)s) * %(similarity_range)s, 1) AND
-            length(event_pattern) <= length(%(target_pattern)s) * (1 + %(similarity_range)s)
-        ORDER BY
-            -- Order by similarity score (lower is more similar)
-            abs(click_count - %(target_clicks)s) +
-            abs(keypress_count - %(target_keypresses)s) +
-            abs(mouse_activity_count - %(target_mouse)s) +
-            abs(active_milliseconds - %(target_active_ms)s) / 1000 ASC
+            length(urls) > 0 AND
+            urls = url_sequence
```

Comment:
so we're just doing exact url sequence match?
i guess reasonable to check speed and matching perf

have you got access to metabase?
we could try and see if we can measure how many sessions per team have matching url sequences
ofc can just ship it and see too :)